qrm/internal/null_types.go

```diff
@@ -111,40 +112,58 @@ type NullUInt64 struct {
 	Valid  bool
 }
 
+// validatePositive validates that the numeric value of any time is a positive entity, safely casts to unsigned type
+func validatePositive[T constraints.Integer](n T) bool {
+	return n > 0
+}
+
 // Scan implements the Scanner interface.
 func (n *NullUInt64) Scan(value interface{}) error {
+	var badState = fmt.Errorf("cannot cast a negative value to an unsigned value, buffer overflow error")
```

Comment:
Error can be moved outside the function, so it can be used inside tests for comparison.